tests/integration/test_methodologies_johnny_decimal_all.py

```
@@ -0,0 +1,1392 @@
+        assert any("Deep hierarchy" in w for w in result.warnings)
+
+    def test_warns_on_many_top_level_folders(self, tmp_path: Path) -> None:
+        from file_organizer.methodologies.johnny_decimal.scanner import FolderScanner
+
+        for i in range(12):
+            (tmp_path / f"folder{i}").mkdir()
+        scanner = FolderScanner()
+        result = scanner.scan_directory(tmp_path)
+        assert any("top-level folders" in w for w in result.warnings)
+
+    def test_looks_like_jd_number_area(self, tmp_path: Path) -> None:
```

Ruff (0.15.9)

[warning] Unused method argument: `tmp_path`

(ARG002)